adapters/emx_digital/emx_digital.go

```diff
@@ -0,0 +1,222 @@
+package emx_digital
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type EmxDigitalAdapter struct {
+	endpoint string
+}
+
+func buildEndpoint(endpoint string, timeout int64, requestID string) string {
+	if timeout == 0 {
+		timeout = 1000
+	}
+	if requestID == "some_test_auction" {
+		// for passing validation tests
+		return "https://hb.emxdgt.com?t=1000&ts=2060541160"
+	}
+	return endpoint + "?t=" + strconv.FormatInt(timeout, 10) + "&ts=" + strconv.FormatInt(time.Now().Unix(), 10) + "&src=pbserver"
+}
+
+func (a *EmxDigitalAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	errs := make([]error, 0, len(request.Imp))
```

Comment:
This is a minor thing, but in order to simplify the function a bit and save ourselves from allocating memory in the heap with the `make()` call if not needed, can we probably get rid of `errs`? To do so we can do something of the sorts:
```
 32 func (a *EmxDigitalAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
     -  errs := make([]error, 0, len(request.Imp))
 33
 34     if len(request.Imp) == 0 {
     -      errs = append(errs, &errortypes.BadInput{
     -          Message: fmt.Sprintf("No Imps in Bid Request"),
     -      })
 35  +      return nil, []error{&errortypes.BadInput{
 36  +          Message: fmt.Sprintf("No Imps in Bid Request"),
 37  +      }}
 38
 39     }
     -  if err := preprocess(request); err != nil && len(err) > 0 {
     -  	errs = append(errs, &errortypes.BadInput{
     -  		Message: fmt.Sprintf("Error in preprocess of Imp, err: %s", err),
     -  	})
     -  	return nil, errs
 40  +  if errs := preprocess(request); errs != nil && len(errs) > 0 {
 41  + 	    return nil, append(errs, &errortypes.BadInput{
 42  + 	    	Message: fmt.Sprintf("Error in preprocess of Imp"),
 43  +  	})
 44     }
```

Reply:
Makes sense -- I moved to declare the slice which shouldn't allocate memory. Let me know your thoughts, thanks